### ISSUE TITLE
Recognize co-authored-by in commit messages

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy-bot#85]
See for example https://github.com/sympy/sympy/pull/19158

**Comment by iammosespaulr** [comment]
https://github.com/sympy/sympy-bot/blob/c600efc24edc6c74fa0837fa5f556dd0aa64f13f/sympy_bot/webapp.py#L82-L91

Searching message in https://github.com/sympy/sympy-bot/blob/c600efc24edc6c74fa0837fa5f556dd0aa64f13f/sympy_bot/webapp.py#L85 for `Co-authored-by: author-name <name@xxx.com>` using regex and adding them to users might work ?

**Comment by iammosespaulr** [comment]
> It would be better if there were direct support for it in the GitHub API. The reason is that we need a GitHub username for the release notes.

Yeah it would! I can't seem to find support for it in the Official API's. 
However, 
`https://api.github.com/search/users?q=iammosespaulr@gmail.com` does provide the username.

**Comment by asmeurer** [comment]
So we can query the email. If we don't get a username from it, we can just include the name and email directly (it doesn't *have* to be a username, that's just the format we use).
